src/Remote/deluserfromgroup/entry.c

```diff
@@ -0,0 +1,106 @@
+#include <windows.h>
+#include <stdio.h>
+#include <lmaccess.h>
+#include "beacon.h"
+#include "bofdefs.h"
+#include "base.c"
+
+
+DWORD DelUserFromDomainGroup(LPCWSTR lpswzServer, LPCWSTR lpswzUserName, LPCWSTR lpswzGroupName)
+{
+	NET_API_STATUS dwErrorCode = NERR_Success;
+	dwErrorCode = NETAPI32$NetGroupDelUser(lpswzServer, lpswzGroupName, lpswzUserName);
+	return dwErrorCode;
+}
+
+DWORD DelUserFromLocalGroup(LPCWSTR lpswzServer, LPCWSTR lpswzUserName, LPCWSTR lpswzGroupName, LPCWSTR lpswzDomainName)
+{
+	NET_API_STATUS dwErrorCode = NERR_Success;
+	LOCALGROUP_MEMBERS_INFO_3 mi[1] = {0}; // https://learn.microsoft.com/en-us/windows/win32/api/lmaccess/ns-lmaccess-localgroup_members_info_3
+	mi[0].lgrmi3_domainandname = intAlloc(1024);
+	if (lpswzDomainName != NULL)
+	{
+		MSVCRT$wcscat(mi[0].lgrmi3_domainandname, lpswzDomainName);
+		MSVCRT$wcscat(mi[0].lgrmi3_domainandname, L"\\");
+	}
+	MSVCRT$wcscat(mi[0].lgrmi3_domainandname, lpswzUserName);
+	dwErrorCode = NETAPI32$NetLocalGroupDelMembers(lpswzServer, lpswzGroupName, 3, (LPBYTE)mi, 1); 
+	return dwErrorCode;
+}
+
+#ifdef BOF
+VOID go( 
+	IN PCHAR Buffer, 
+	IN ULONG Length 
+) 
+{
+	DWORD dwErrorCode = ERROR_SUCCESS;
+	datap parser = {0};
+	BeaconDataParse(&parser, Buffer, Length);
+	LPCWSTR lpswzDomainName = (LPCWSTR)BeaconDataExtract(&parser, NULL); // $5
+	LPCWSTR lpswzHostName = (LPCWSTR)BeaconDataExtract(&parser, NULL); // $4
+	LPCWSTR lpswzUserName = (LPCWSTR)BeaconDataExtract(&parser, NULL); // $2
+	LPCWSTR lpswzGroupName = (LPCWSTR)BeaconDataExtract(&parser, NULL);// $3
+	if(lpswzHostName[0] == L'\0'){lpswzHostName = NULL;}
+	if(lpswzDomainName[0] == L'\0'){lpswzDomainName = NULL;}
+
+	if(!bofstart())
+	{
+		return;
+	}
+
+	if (lpswzDomainName == NULL && lpswzHostName != NULL)

```

Comment:
In the description all of the available arguments are listed as required

If I give both a hostname and a domain name, SUCCESS. will be printed but nothing will happen.

I'd split this into two aliases one to remove a user from a computer local group and one to remove it from a domain group.  Then you can enforce this logic in the cna script easily.

barring that the cna shouldn't list everything as required and should explain what is expected for local vs domain group deletion.